### DISCUSSION TITLE
Peptide design

### DISCUSSION THREAD

**Discussion by JamesRandallJR** [RosettaCommons/rosetta#174] [General]
Using Rosetta to design peptides, the general outlines can differ as the software provides different tools.
Is it better to have backbones and build on the sequences, then have a high throughput docking for the generated peptides or I should follow the anchor extension approach for designing?

Also for the first method, is FlexPepDock the only applicable executable for peptide docking?
And for anchor extension, if I have a target structure with a small molecule bound to it or even don't have a binder at all, how I shall make it work? 

Detailed info is welcomed.

**Comment by roccomoretti** [comment]
FlexPepDock is one of the main method for peptide docking in Rosetta. There's certainly other protocols one could use for docking peptides, but they aren't necessarily tailored to peptide docking (e.g. RosettaDock assumes more-or-less fixed backbone structure when docking. RosettaLigand would treat the peptide as a small molecule and would potentially have issues with the number of rotatable bonds for typical peptides.)  If you're looking specifically at cyclic peptides, [Mulligan and Hosseinzadeh](https://pubmed.ncbi.nlm.nih.gov/?term=mulligan+vk+Hosseinzadeh+P) have put together some alternative sampling protocols which take advantage of the constrained nature of (N-to-C) cyclized peptides.

Regarding design approach, it really depends on the details of your system, how the sequence/backbone landscape looks like, and how much information you want to pre-specify about the design. The general approach is to define/create a backbone conformation and then sample the sequence on top of that (fixed) backbone conformation. How you approach the backbone creation and the sequence sampling -- and whether you do it as a once-through or as an iterative approach -- depends a bit on how much information about your system you can pre-specify, and how you want to expand from the known information to the generated information.

If you already have a good idea about where the backbone "should" be, then using the pre-specified backbone and build the sequences on top of it might be a better approach (as the design will be tailored to the pre-specified backbone). Note that this isn't necessarily a one-shot case: after you have a preliminary design, you can attempt to diversify or resampe the backbone (e.g. by redocking) and then do more rounds of design/redocking until you get reasonable convergence or a set of designs you're happy with.

Anchor extension works well if you do have a starting anchor position (a key residue-receptor interaction) and you're looking to expand outward from it. It will keep the anchor residue interaction more-or-less stable while progressively sampling larger peptides. As before, you can proceed from the generated backbones into an iterative refinement if you so desire.

A bit of a more complicated situation is if you don't have a pre-specified backbone or any pre-defined interactions you want to preserve. It's in this case you'd probably want to take a more docking-based approach, where you sample for likely backbone conformations (or residue interactions) with a preliminary docking, and then when you find one or more backbone conformations (or residue interactions) you can take that information forward into other approaches.

I'll note that the Mulligan and Hosseinzadeh approach takes a similar sort of approach. In certain cases it first pre-enumerates likely peptide backbone conformations in the absence of a binding partner, then it docks the (now pre-specified) cyclic backbone conformations into the binding partner, and then does design to optimize the interaction -- note this process is highly simplified. Depending on the context you introduce design and filtering steps between other steps, and repeat the design/redocking/backbone resampling process to refine designs.

Regarding the case where you have a target structure with a small molecule bound, the question is how similar that small molecule is to the peptide you want to design, and how much information from the protein/small molecule interaction you can leverage. If the interaction is very amino acid like, then you can (manually) replace the small molecule with the corresponding amino acid making the similar sorts of interactions, and the proceed from anchor extension with the imposed residue. If the interactions can't be recapitulated by an amino acid, then there really isn't enough information for anchor extension _per se_ to be successful. (The point of anchor extension being to build a peptide while keeping pre-existing interactions -- you don't really have any pre-existing interactions to keep.) At best the information you can extract from the ligand is the likely binding pocket, which is helpful for the more general design process if not for anchor extension.

**Reply by JamesRandallJR** [reply]
Thanks for the comprehensive answer
